pkgs/top-level/all-packages.nix

```diff
@@ -12970,10 +12970,12 @@ with pkgs;
      for a specific kernel.  This function can then be called for
      whatever kernel you're using. */
 
-  linuxPackagesFor = kernel: lib.makeExtensible (self: with self; {
+  linuxPackagesFor = kernel: let stdenvKernelCc = overrideCC stdenv kernel.buildCc;
```

Comment:
I haven't found a better way to express that other than with introducing a "let" here.

Reply:
What about just `stdenv = overrideCC pkgs.stdenv kernel.buildCc;`?